### PR TITLE
Respect `activate` value for Functions Deployments

### PR DESCRIPTION
**Pull request by ItzNotABug**
## What does this PR do?

This PR fixes the issue where the if `activate` is false when deploying a function, it is still activated regardless.

## Test Plan

Manual, see -

https://github.com/appwrite/appwrite/assets/20625965/26e4abb6-6013-4059-b7a4-49edcca20500

## Related PRs and Issues

- NA

## Checklist

- [x] Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?
- [x] If the PR includes a change to an API's metadata (desc, label, params, etc.), does it also include updated API specs and example docs?

### CODE REVIEW

File: app/controllers/api/functions.php [hunk 1]
```diff
@@ -1050,7 +1050,7 @@
     ->param('entrypoint', null, new Text(1028), 'Entrypoint File.', true)
     ->param('commands', null, new Text(8192, 0), 'Build Commands.', true)
     ->param('code', [], new File(), 'Gzip file with your code package. When used with the Appwrite CLI, pass the path to your code directory, and the CLI will automatically package your code. Use a path that is within the current directory.', skipValidation: true)
-    ->param('activate', false, new Boolean(true), 'Automatically activate the deployment when it is finished building.')
+    ->param('activate', false, new Boolean(true), 'Activate the deployment when it is finished building.', skipValidation: true)
```
Comment (Meldiron) [inline app/controllers/api/functions.php:1053]: Is `skipValidation` nessessary addition? Does it not work without it - what's the error?

Comment (ItzNotABug) [reply app/controllers/api/functions.php:1053]: It doesn't actually. The issue is `('sdk.request.type', 'multipart/form-data')` actually which transforms the `false` value to an empty string (`''`) while the truthy ones are properly converted to `1`. This triggers a validation failure from the `new Boolean(true)`.

**Comment by ItzNotABug** [comment]
The latest commit 269cdc3 properly uses the correct status-code (the only failing test), a re-run of the tests might not be necessary considering all of them run for quite a while.

**Comment by ItzNotABug** [comment]
Closing in favour of - https://github.com/appwrite/appwrite/pull/8358.

**Pull request closed without merging**